cmd/sync/sync.go

```diff
@@ -3,64 +3,232 @@
 package main
 
 import (
+	"encoding/json"
 	"io"
+	"os"
 
+	"sda-pipeline/internal/broker"
 	"sda-pipeline/internal/config"
+	"sda-pipeline/internal/database"
 	"sda-pipeline/internal/storage"
 
 	log "github.com/sirupsen/logrus"
+	"github.com/xeipuuv/gojsonschema"
 )
 
-type trigger struct {
-	Type     string `json:"type"`
-	User     string `json:"user"`
-	Filepath string `json:"filepath"`
+// sync struct that holds the json message data
+type sync struct {
+	Type               string      `json:"type"`
+	User               string      `json:"user"`
+	Filepath           string      `json:"filepath"`
+	AccessionID        string      `json:"accession_id"`
+	DecryptedChecksums []checksums `json:"decrypted_checksums"`
+}
+
+// Checksums is struct for the checksum type and value
+type checksums struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+// Completed is struct holding the full message data
+type completed struct {
+	User               string      `json:"user"`
+	Filepath           string      `json:"filepath"`
+	AccessionID        string      `json:"accession_id"`
+	DecryptedChecksums []checksums `json:"decrypted_checksums"`
 }
 
 func main() {
-	conf, err := config.NewConfig("ingest")
+	conf, err := config.NewConfig("sync")
 	if err != nil {
 		log.Fatal(err)
 	}
-
-	archive, err := storage.NewBackend(conf.Archive)
+	mq, err := broker.NewMQ(conf.Broker)
 	if err != nil {
 		log.Fatal(err)
-
 	}
+	db, err := database.NewDB(conf.Database)
+	if err != nil {
+		log.Fatal(err)
+	}
+	backup, err := storage.NewBackend(conf.Backup)
+	if err != nil {
+		log.Fatal(err)
 
+	}
 	inbox, err := storage.NewBackend(conf.Inbox)
```

Comment:
This should be `Archive` rather?

Reply:
I am not sure about that. Maybe @jbygdell has some input?

Reply:
Should be archive or the bucket and access keys will be wrong.

For this service `archive` is the source and `backup` the destination